dataset_loaders/videos/movingMNIST.py

```diff
@@ -0,0 +1,255 @@
+        trajectory_x = (canvas_size[1] * trajectory_x).astype(np.int32)
+        trajectory_y = (canvas_size[0] * trajectory_y).astype(np.int32)
+        return trajectory_x, trajectory_y
+
+    def _get_sequence(self, verbose=False):
+        trajectory_x, trajectory_y = self._get_random_trajectory()
+
+        # minibatch data
+        if self.random_background:
+            out_sequence = self._rng.rand(self.seq_length+1,
+                                          self.frame_size[0],
+                                          self.frame_size[1], 1)
+        else:
+            out_sequence = np.zeros((self.seq_length+1,
+                                     self.frame_size[0],
+                                     self.frame_size[1], 1),
+                                    dtype=np.float32)
+
+        for digit_id in range(self.num_digits):
+
+            # get random digit from dataset
+            curr_data_idx = self._rng.randint(
+                0, self._MNIST_data.shape[0]-1)
+            digit_image = self._MNIST_data[curr_data_idx]
+
+            zoom_factor = 1
+            if self.digits_sizes != 28:
+                zoom_factor = int(self.digits_sizes[digit_id]/28)
+                digit_image = zoom(digit_image, zoom_factor)
+            digit_size = digit_image.shape[0]
+
+            # generate video
+            digit_image = np.expand_dims(digit_image, -1)
+            import ipdb; ipdb.set_trace()
+            for i in range(self.seq_length+1):
+                top = trajectory_y[i, digit_id]
+                left = trajectory_x[i, digit_id]
+                bottom = top + digit_size
+                right = left + digit_size
+                out_sequence[i, top:bottom, left:right, :] = np.maximum(
+                    out_sequence[i, top:bottom, left:right, :], digit_image)
+        return out_sequence
+
+    def _reset(self, *args, **kwargs):
+        """ This function is used to reset the rng and data index from
+        where the digits are sampled to create the video sequence
+        """
+        if self.which_set != 'train':
+            self._rng = copy.deepcopy(self._origin_rng)
+
+    def _fill_names_batches(self, *args, **kwargs):
+        # Reset the digit generator at the end of the epoch
+        self._reset(self, *args, **kwargs)
+        super(MovingMNISTDataset, self)._fill_names_batches(*args, **kwargs)
+
+    def load_sequence(self, sequence):
+        """Load a sequence of images/frames
+
+        Auxiliary function that loads a sequence of frames with
+        the corresponding ground truth and their filenames.
+        Returns a dict with the images in [0, 1], their corresponding
+        labels, their subset (i.e. category, clip, prefix) and their
+        filenames.
+        """
+        F = []
+        for frame_name, _ in sequence:
+            F.append(frame_name)
+        sequence = self._get_sequence()
+        X = sequence[:self.seq_length]
+        Y = sequence[self.seq_length]
```

Comment:
replace self.seq_length with `-1` in both lines. It's equivalent but more directly comprehensible IMO